packages/core-mobile/app/new/components/SlideToConfirm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useRef } from 'react'
+import { PanResponder, Animated, Dimensions } from 'react-native'
+import { useTheme, View, Icons } from '@avalabs/k2-alpine'
+
+const { width } = Dimensions.get('window')
+const _sliderWidth = 64
+const _sliderHeight = 64
+
+const SlideToConfirm = ({
+  text,
+  onConfirm
+}: {
+  text: string
+  onConfirm: () => void
+}): JSX.Element => {
+  const { theme } = useTheme()
+  const slideWidth = width - 40 // Space for the slider to move
+  const unlockThreshold = slideWidth - _sliderWidth // Set unlock point near the end
+  const sliderWidth = useRef(new Animated.Value(_sliderWidth)).current
+
+  const textColorAnim = sliderWidth.interpolate({
+    inputRange: [0, unlockThreshold],
+    outputRange: theme.isDark
+      ? [theme.colors.$textPrimary, '#000000']
+      : [theme.colors.$textPrimary, '#FFFFFF'],
```

Comment:
I wonder if we can just use theme.colors.$surfacePrimary, so that we can simply do 

`[theme.colors.$textPrimary, heme.colors.$surfacePrimary]`

Reply:
fixed.